samples/js/picasa/slurp/src/main.js

```diff
@@ -0,0 +1,199 @@
+// @flow
+
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+import argv from 'yargs';
+import {
+  DatasetSpec,
+  jsonToNoms,
+  newStruct,
+} from '@attic/noms';
+import {
+  default as fetch,
+  Request,
+  Headers,
+} from 'node-fetch';
+import type {
+  Api,
+  Albums,
+  Faces,
+  User,
+} from './types.js';
+import {getRefreshToken, getAccessTokenFromRefreshToken} from './oauth.js';
+
+const args = argv
+  .usage(
+    'Parses photo metadata out of Picasa API\n\n' +
+    'Usage: either provide:\n' +
+    ' 1. an access token (these are short lived, and should only be used for testing)\n' +
+    '    you can get these from https://developers.google.com/oauthplayground/\n' +
+    ' 2. a client ID and secret, with an optional refresh token\n' +
+    'If 2 and if a refresh token isn\'t provided, one will be fetched, then picasa-slurp\n' +
+    'must be re-run with that refresh token. Hold on to it.\n\n' +
+    'In other words:\n' +
+    ' node . --access-token <dest-dataset>\n' +
+    'or\n' +
+    ' node . --client-id=<id> --client-secret=<secret> [--refresh-token=<token> <dest-dataset>]\n')
+  .option('access_token', {
+    describe: 'OAuth2 access token (these are short lived and should only be used for testing)',
+    type: 'string',
+  })
+  .option('client_id', {
```

Comment:
@aboodman could you do this in a follow up? I don't have time (literally only have 30 seconds)